modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/addimages/plugin.js

```diff
@@ -0,0 +1,240 @@
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _onDragDrop
+		 * @param {CKEDITOR.dom.event} event dragdrop event, as received natively from CKEditor
+		 * @protected
+		 */
+		_onDragDrop(event) {
+			const nativeEvent = event.data.$;
+
+			const transferFiles = nativeEvent.dataTransfer.files;
+
+			if (transferFiles.length > 0) {
+				new CKEDITOR.dom.event(nativeEvent).preventDefault();
+
+				const editor = event.editor;
+
+				this._handleFiles(transferFiles, editor);
+			}
+		},
+
+		/**
+		 * Handles drag enter event. In case of IE, this function will prevent the event.
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _onDragEnter
+		 * @param {DOM event} event dragenter event, as received natively from CKEditor
+		 * @protected
+		 */
+		_onDragEnter(event) {
+			if (isIE) {
+				this._preventEvent(event);
+			}
+		},
+
+		/**
+		 * Handles drag over event. In case of IE, this function will prevent the event.
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _onDragOver
+		 * @param {DOM event} event dragover event, as received natively from CKEditor
+		 * @protected
+		 */
+		_onDragOver(event) {
+			if (isIE) {
+				this._preventEvent(event);
+			}
+		},
+
+		/**
+		 * Checks if the pasted data is image and passes it to
+		 * {{#crossLink "CKEDITOR.plugins.addimages/_processFile:method"}}{{/crossLink}} for processing.
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _onPaste
+		 * @param {CKEDITOR.dom.event} event A `paste` event, as received natively from CKEditor
+		 * @protected
+		 */
+		_onPaste(event) {
+			if (
+				event.data &&
+				event.data.$ &&
+				event.data.$.clipboardData &&
+				event.data.$.clipboardData.items &&
+				event.data.$.clipboardData.items.length > 0
+			) {
+				const pastedData = event.data.$.clipboardData.items[0];
+
+				if (pastedData.type.indexOf('image') === 0) {
+					const imageFile = pastedData.getAsFile();
+
+					this._processFile(imageFile, event.editor);
+				}
+			}
+		},
+
+		/**
+		 * Prevents a native event.
+		 *
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _preventEvent
+		 * @param {DOM event} event The event to be prevented.
+		 * @protected
+		 */
+		_preventEvent(event) {
+			event = new CKEDITOR.dom.event(event.data.$);
+
+			event.preventDefault();
+			event.stopPropagation();
+		},
+
+		/**
+		 * Processes an image file. The function creates an img element and sets as source
+		 * a Data URI, then fires an 'imageAdd' event via CKEditor's event system.
+		 *
+		 * @fires CKEDITOR.plugins.addimages#imageAdd
+		 * @instance
+		 * @memberof CKEDITOR.plugins.addimages
+		 * @method _preventEvent
+		 * @param {DOM event} event The event to be prevented.
+		 * @protected
+		 */
+		_processFile(file, editor) {
+			const reader = new FileReader();
+
+			reader.addEventListener('loadend', () => {
```

Comment:
I'm not sure exactly how these `_diffs` files get integrated into the build and if there are any discrepancies from the normal process. Can you confirm that these arrow functions get transpiled in the `classes/` output, just to be sure that we don't break IE?

Reply:
Hey @wincent,

I'm not a gradle expert, but apparently the `_diffs` directory [is copied](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/build.gradle#L48-L56) and then everything is built as the other modules.

Here's a screenshot of the transpiled code in `classes`:
![screenshot](https://user-images.githubusercontent.com/5572/89387747-37d80880-d703-11ea-9a47-ceae81f64806.png)